src/server/pps/server/api_server.go

```diff
@@ -1422,6 +1408,48 @@ func job(jobInfo *persist.JobInfo) *batch.Job {
 	}
 }
 
+func (a *apiServer) deletePipeline(ctx context.Context, pipeline *ppsclient.Pipeline) error {
+	persistClient, err := a.getPersistClient()
+	if err != nil {
+		return err
+	}
+
+	if pipeline == nil {
+		return fmt.Errorf("Pipeline cannot be nil")
+	}
+
+	// Delete kubernetes jobs.  Otherwise we won't be able to create jobs with
+	// the same IDs, since kubernetes jobs simply use these IDs as their names
+	jobInfos, err := persistClient.ListJobInfos(ctx, &ppsclient.ListJobRequest{
+		Pipeline: pipeline,
+	})
+	if err != nil {
+		return err
+	}
+	for _, jobInfo := range jobInfos.JobInfo {
+		if err = a.kubeClient.Extensions().Jobs(a.namespace).Delete(jobInfo.JobID, nil); err != nil {
+			return err
```

Comment:
Makes sense to me. I actually didn't write this code as part of the PR, just relocated it from `DeletePipeline` so I'm not sure if there was some reason why returning the error here was correct. But it certainly seems like just logging it would be better so I'll do that.